main.go

```diff
@@ -372,8 +372,8 @@ func main() {
 	}
 
 	// This is kind of expensive because it looks at the state and formats text
-	// output on an ongoing basis. Only run in debug mode.
-	if config.Sidecar.Debug {
+	// output on an ongoing basis. Only run in debug logging mode.
+	if config.Sidecar.LoggingLevel == "debug" {
```

Comment:
This, it turns out was a HUGE cost of memory allocs and CPU. It was doing all this work to generate the output and then throwing it away when not in debug level. This reduces CPU a huge amount.